compiler/rustc_abi/src/lib.rs

```diff
@@ -1485,12 +1491,12 @@ impl BackendRepr {
             BackendRepr::ScalarPair(s1, s2) => {
                 // May have some padding between the pair.
                 let field2_offset = s1.size(cx).align_to(s2.align(cx).abi);
-                (field2_offset + s2.size(cx)).align_to(self.inherent_align(cx)?.abi)
+                (field2_offset + s2.size(cx)).align_to(self.required_align(cx)?)
```

Comment:
Something seems *really wrong* if we can't compute the offset of the 2nd field of a ScalarPair. I would have expected an `unwrap`. But I guess this is pre-existing so :shrug: I don't understand any of this I guess.^^

I originally wrote the layout sanity check, but not because I understand layouts particularly well -- rather, because I don't understand them at all and wanted to see some structure.